service/worker/archiver/activities.go

```diff
@@ -23,80 +23,74 @@ package archiver
 import (
 	"context"
 	"errors"
-	"time"
 
 	"github.com/uber/cadence/.gen/go/shared"
 	"github.com/uber/cadence/common"
+	carchiver "github.com/uber/cadence/common/archiver"
 	"github.com/uber/cadence/common/backoff"
-	"github.com/uber/cadence/common/blobstore"
-	"github.com/uber/cadence/common/blobstore/blob"
-	"github.com/uber/cadence/common/cache"
 	"github.com/uber/cadence/common/log/tag"
 	"github.com/uber/cadence/common/metrics"
 	"github.com/uber/cadence/common/persistence"
 	"go.uber.org/cadence"
 	"go.uber.org/cadence/activity"
 )
 
-type (
-	uploadProgress struct {
-		UploadedBlobs   []string
-		IteratorState   []byte
-		BlobPageToken   int
-		HandledLastBlob bool
-	}
-
-	uploadResult struct {
-		BlobsToDelete    []string
-		ErrorWithDetails string
-	}
-)
-
 const (
 	uploadHistoryActivityFnName = "uploadHistoryActivity"
-	deleteBlobActivityFnName    = "deleteBlobActivity"
 	deleteHistoryActivityFnName = "deleteHistoryActivity"
-	blobstoreTimeout            = 30 * time.Second
-
-	errInvalidRequest = "archival request is invalid"
-	errGetDomainByID  = "could not get domain cache entry"
-	errConstructKey   = "could not construct blob key"
-	errGetTags        = "could not get blob tags"
-	errUploadBlob     = "could not upload blob"
-	errReadBlob       = "could not read blob"
-	errEmptyBucket    = "domain is enabled for archival but bucket is not set"
-	errConstructBlob  = "failed to construct blob"
-	errDownloadBlob   = "could not download existing blob"
-	errDeleteBlob     = "could not delete existing blob"
 
 	errDeleteHistoryV1 = "failed to delete history from events_v1"
 	errDeleteHistoryV2 = "failed to delete history from events_v2"
 
-	errHistoryMutated = "history was mutated during uploading"
-
-	errActivityPanic = "cadenceInternal:Panic"
+	errActivityPanic       = "cadenceInternal:Panic"
+	errTimeoutStartToClose = "cadenceInternal:Timeout START_TO_CLOSE"
```

Comment:
I don't think you need to include timeout errors as non-retryable errors - those are seen as the workflow getting an error rather than the activity returning an error.